Assets/Scripts/Weapons/Weapon.cs

```diff
@@ -0,0 +1,68 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class Weapon : MonoBehaviour
+{
+    GameObject cameraView;
+
+    Dictionary<string, bool> _firingBehavior = new Dictionary<string, bool>()
+    {
+        { "singleShot", true }, // singleShot == !repeater
+        { "shotgun", false },
+        { "repeater", false }
+    };
+
+    IHitBehaviour _hitBehaviour;
+    float _fireRate; // shots per second
+    int _magazineSize;
+    int _currentMagazine;
+    double lastShot = 0;
+
+    Dictionary<string, bool> _hitTypeModifiers = new Dictionary<string, bool>()
+    {
+        { "exploding", false },
+        { "pierceThrough", false }
+    };
+    Dictionary<string, float> _hitStatsModifiers = new Dictionary<string, float>()
+    {
+        { "damage", 10f },
+        { "bulletSpeed", 20f }
+    };
+
+    void Start()
+    {
+        cameraView = transform.GetChild(0).gameObject;
+
+        _hitBehaviour = new BulletBehaviour();
+        _fireRate = 0.1f; 
+        _magazineSize = 10;
+        _currentMagazine = _magazineSize;
+    }
+
+    public void Attack()
+    {
+        if (Time.time - lastShot > _fireRate)
```

Comment:
Attack should take a parameter, direction, instead of determining its own aiming direction. This is to decouple aiming and firing.

Reply:
https://github.com/MaXeraph/Adrenarun/pull/4/commits/d65f1bb4fc4d7b6c9c127e9824b438d75c8e0b50#diff-37fbd14d03c95577256eae4355cb9cb95b45222f5570294b5aac806986183b0aR28